packages/core/src/components/fields/ArrayField.js

```diff
@@ -393,6 +404,31 @@ class ArrayField extends Component {
     };
   };
 
+  onCopyIndexClick = index => {
+    return event => {
+      if (event) {
+        event.preventDefault();
+      }
+      const { onChange } = this.props;
+      const { keyedFormData } = this.state;
+      const newKeyedFormDataRow = {
+        key: generateRowId(),
+        item: keyedFormData[index].item,
+      };
+      const newKeyedFormData = [
+        ...this.state.keyedFormData,
+        newKeyedFormDataRow,
+      ];
```

Comment:
I see that you're always adding the item at the end of the array, but maybe it would make sense to give the user the option to add right after the element that you're copying. Maybe the function `onCopyIndexClick` could receive a second argument to tell where you want to add the element (could be an index of insertion or an enum)